Shared/Realm.Shared/Handles/LinkListHandle.cs

```diff
@@ -21,7 +21,7 @@
 
 namespace Realms
 {
-    internal class LinkListHandle : RealmHandle
+    internal class LinkListHandle : CollectionHandleBase
```

Comment:
Because this now wraps the Object Store `List` class, I think we should rename this to just `ListHandle`.